vunit/vhdl_parser.py

```diff
@@ -988,16 +988,15 @@ def reference_all_names_within(self):
         return self.name_within == "all"
 
 
-VHDL_REMOVE_COMMENT_RE = re.compile(r'--[^\n]*')
+VHDL_REMOVE_COMMENT_RE = re.compile(r'^([^\"\-]*(?:\"[^\"]*\"[^\"\-]*)*)--.*')
```

Comment:
```suggestion
VHDL_REMOVE_COMMENT_RE = re.compile(r'^([^\"\-]*(?:\"(?:\"\"|[^\"])*\"[^\"\-]*)*)--.*')
```
A correct implementation would look like this, since VHDL allows to escape the `"` inside a string literal with `""`. So `""""` means `"`.